core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -274,15 +275,47 @@ private void extract(PluginHandle pluginHandle, Terminal terminal, Path pluginFi
             try {
                 terminal.println(VERBOSE, "Found config, moving to %s", destConfigDirectory.toAbsolutePath());
                 moveFilesWithoutOverwriting(sourceConfigDirectory, destConfigDirectory, ".new");
+
+                if (Environment.getFileStore(destConfigDirectory).supportsFileAttributeView(PosixFileAttributeView.class)) {
+                    //We take owner, group and permissions from the parent ES_CONFIG directory, assuming they were properly set depending
+                    // on how es was installed in the first place: can be root:elasticsearch (750) if es was installed from rpm/deb packages
+                    // or elasticsearch:elasticsearch if installed from tar/zip. As for permissions we don't rely on umask.
+                    PosixFileAttributes parentDirAttributes = Files.getFileAttributeView(destConfigDirectory.getParent(), PosixFileAttributeView.class).readAttributes();
+                    Files.walkFileTree(destConfigDirectory, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            if (attrs.isRegularFile()) {
+                                setPosixFileAttributes(file, parentDirAttributes);
```

Comment:
good catch, I think it will set execute on files if the parent dir has it set, but probably shouldn't. For subdirectories, setting execute is what we want but on the config files themselves we shouldn't set +x. Make sense?